src/Tribe/JSON_LD/Abstract.php

```diff
@@ -126,18 +154,79 @@ public function markup( $post = null, $args = array() ) {
 
 		/**
 		 * Allows users to filter the end markup of JSON-LD
+		 *
 		 * @deprecated
 		 * @todo Remove on 4.4
+		 *
 		 * @param string The HTML for the JSON LD markup
 		 */
 		$html = apply_filters( 'tribe_google_data_markup_json', $html );
 
 		/**
 		 * Allows users to filter the end markup of JSON-LD
+		 *
 		 * @param string The HTML for the JSON LD markup
 		 */
 		$html = apply_filters( 'tribe_json_ld_markup', $html );
 
 		echo $html;
 	}
+
+	/**
+	 * @return array An array of post IDs.
+	 */
+	public function get_fetched_post_ids() {
+		return self::$fetched_post_ids;
+	}
+
+	/**
+	 * Marks a post as already fetched.
+	 *
+	 * @param int|WP_Post $post
+	 */
+	public function set_fetched_post_id( $post ) {
+		$post = $this->get_post_object( $post );
+
+		if ( empty( $post ) || in_array( $post->ID, self::$fetched_post_ids ) ) {
+			return;
+		}
+
+		self::$fetched_post_ids[] = $post->ID;
+	}
+
+	public function reset_fetched_post_ids() {
+		self::class_reset_fetched_post_ids();
+	}
+
+	public function unset_fetched_post_id( $post ) {
+		$post = $this->get_post_object( $post );
```

Comment:
We're really only interested in getting the ID here: is it possible to drop the `get_post_object()` method and call `Tribe__Main::post_id_helper()` directly?